compiler/src/type_check/define_types.rs

```diff
@@ -1996,4 +1996,73 @@ mod tests {
         assert_eq!(error.id(), DiagnosticId::InvalidType);
         assert_eq!(error.location(), &cols(16, 20));
     }
+
+    #[test]
+    fn test_check_class_with_trait_and_default_method() {
```

Comment:
Testing this in Rust is going to be a pain, as you'd probably need to run all compiler passes that are run as part of type checking.

I've been meaning to add some code such that we can write these tests in Inko itself, so I'll take a look at that this wekend.

Reply:
Great to know! Thanks! In the meantime I'll look into some other issues.

Reply:
@bartlomieju fa7915c216334bfebd03a079d79e1a0213fe4b1a adds a setup that should make it (much) easier to write diagnostics tests. You can see an example [here](https://github.com/inko-lang/inko/blob/main/std/test/diagnostics/duplicate_method.inko). You can read up a bit more about how to work with such tests [here](https://docs.inko-lang.org/manual/main/guides/contributing/#compiler-diagnostic-tests).

Reply:
Sweet! I updated the PR to include the test - I verified that it fails on `main` but passes on this branch. Thanks!